python/lsst/qserv/tests/benchmark.py

```diff
@@ -305,40 +325,62 @@ def run(self, mode_list, load_data, stop_at_query=MAX_QUERY):
                 _LOG.info("Tables to Duplicate %s", self.dataReader.duplicatedTables)
                 self.dataDuplicator.run()
 
-        for mode in mode_list:
-
-            dbName = "qservTest_case%s_%s" % (self._case_id, mode)
-
-            if load_data:
+        if load_data:
+            # when loading qserv_async is the same as qserv (do not load twice)
+            load_modes = set('qserv' if mode == 'qserv_async' else mode for mode in mode_list)
+            for mode in load_modes:
+                dbName = "qservTest_case%s_%s" % (self._case_id, mode)
                 self.loadData(mode, dbName)
 
+        for mode in mode_list:
+
+            dbName = "qservTest_case%s_%s" % (self._case_id,
+                                              'qserv' if mode == 'qserv_async' else mode)
             self.runQueries(mode, dbName, stop_at_query)
 
-    def analyzeQueryResults(self):
+    def analyzeQueryResults(self, mode_list):
         """Compare results from runs with different modes.
+
+        If "mysql" is in the mode_list compare all other modes against "mysql",
+        otherwise compare all others against first.
+
+        Parameters
+        ----------
+        mode_list : list
+            List of strings like "mysql", "qserv", length of list should be at least 2.
         """
 
         outputs_dir = os.path.join(self._out_dirname, "outputs")
 
         failing_queries = []
 
-        mysql_out_dir = os.path.join(outputs_dir, "mysql")
-        qserv_out_dir = os.path.join(outputs_dir, "qserv")
+        if 'mysql' in mode_list:
+            baseline = 'mysql'
+        else:
+            baseline = mode_list[0]
+
+        baseline_out_dir = os.path.join(outputs_dir, baseline)
+        for mode in mode_list:
+            if mode == baseline:
+                continue
```

Comment:
It might be more clear to remove the item from mode_list (or create a new derivative list with the item removed) on lines 358/360, but if so it's just a nit.